File: native/macos/SkillFlow/Sources/Prompts/PromptsModels.swift
import Foundation

struct PromptEntry: Codable, Identifiable, Equatable {
    let name: String
    let description: String
    let category: String
    let path: String
    let filePath: String
    let content: String
    let imageURLs: [String]
    let webLinks: [PromptLink]
    let createdAt: Date
    let updatedAt: Date

    var id: String { name }
}

struct PromptLink: Codable, Equatable {
    let label: String
    let url: String
}

struct PromptNameParams: Encodable {
    let name: String
}

struct PromptMoveCategoryParams: Encodable {
    let name: String
    let category: String
}

struct PromptCategoryNameParams: Encodable {
    let name: String
}

struct PromptRenameCategoryParams: Encodable {
    let oldName: String
    let newName: String
}

struct PromptCreateParams: Encodable {
    let name: String
    let description: String
    let category: String
    let content: String
    let imageURLs: [String]
    let webLinksMarkdown: String
}

struct PromptUpdateParams: Encodable {
    let originalName: String
    let name: String
    let description: String
    let category: String
    let content: String
    let imageURLs: [String]
    let webLinksMarkdown: String
}

struct PromptExportByNamesParams: Encodable {
    let names: [String]
}

File: native/windows/SkillFlow/SkillFlow/Prompts/PromptsModels.cs
using System.Text.Json.Serialization;

namespace SkillFlow.Prompts;

public sealed class PromptEntry
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("description")]
    public string Description { get; set; } = string.Empty;

    [JsonPropertyName("category")]
    public string Category { get; set; } = string.Empty;

    [JsonPropertyName("path")]
    public string Path { get; set; } = string.Empty;

    [JsonPropertyName("filePath")]
    public string FilePath { get; set; } = string.Empty;

    [JsonPropertyName("content")]
    public string Content { get; set; } = string.Empty;

    [JsonPropertyName("imageURLs")]
    public List<string> ImageURLs { get; set; } = new();

    [JsonPropertyName("webLinks")]
    public List<PromptLink> WebLinks { get; set; } = new();

    [JsonPropertyName("createdAt")]
    public DateTimeOffset CreatedAt { get; set; }

    [JsonPropertyName("updatedAt")]
    public DateTimeOffset UpdatedAt { get; set; }

    public string WebLinksText => WebLinks.Count == 0 ? "" : string.Join(", ", WebLinks.Select(l => $"[{l.Label}]({l.URL})"));
}

public sealed class PromptLink
{
    [JsonPropertyName("label")]
    public string Label { get; set; } = string.Empty;

    [JsonPropertyName("url")]
    public string URL { get; set; } = string.Empty;
}

public sealed class PromptNameParams
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;
}

public sealed class PromptMoveCategoryParams
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("category")]
    public string Category { get; set; } = string.Empty;
}

public sealed class PromptCategoryNameParams
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;
}

public sealed class PromptRenameCategoryParams
{
    [JsonPropertyName("oldName")]
    public string OldName { get; set; } = string.Empty;

    [JsonPropertyName("newName")]
    public string NewName { get; set; } = string.Empty;
}

public sealed class PromptCreateParams
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("description")]
    public string Description { get; set; } = string.Empty;

    [JsonPropertyName("category")]
    public string Category { get; set; } = string.Empty;

    [JsonPropertyName("content")]
    public string Content { get; set; } = string.Empty;

    [JsonPropertyName("imageURLs")]
    public List<string> ImageURLs { get; set; } = new();

    [JsonPropertyName("webLinksMarkdown")]
    public string WebLinksMarkdown { get; set; } = string.Empty;
}

public sealed class PromptUpdateParams
{
    [JsonPropertyName("originalName")]
    public string OriginalName { get; set; } = string.Empty;

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("description")]
    public string Description { get; set; } = string.Empty;

    [JsonPropertyName("category")]
    public string Category { get; set; } = string.Empty;

    [JsonPropertyName("content")]
    public string Content { get; set; } = string.Empty;

    [JsonPropertyName("imageURLs")]
    public List<string> ImageURLs { get; set; } = new();

    [JsonPropertyName("webLinksMarkdown")]
    public string WebLinksMarkdown { get; set; } = string.Empty;
}

public sealed class PromptExportByNamesParams
{
    [JsonPropertyName("names")]
    public List<string> Names { get; set; } = new();
}
